common/app/assets/assets.scala

```diff
@@ -1,168 +1,135 @@
+    project match {
+      case "facia" => "stylesheets/ie9.head.facia.css"
+      case "identity" => "stylesheets/ie9.head.identity.css"
+      case "football" => "stylesheets/ie9.head.football.css"
+      case "index" => "stylesheets/ie9.head.index.css"
+      case _ => "stylesheets/ie9.head.content.css"
     }
 
-    private def cssOldIE(project: String): String = {
-      project match {
-        case "facia" => "stylesheets/old-ie.head.facia.css"
-        case "identity" => "stylesheets/old-ie.head.identity.css"
-        case "football" => "stylesheets/old-ie.head.football.css"
-        case "index" => "stylesheets/old-ie.head.index.css"
-        case _ => "stylesheets/old-ie.head.content.css"
-      }
-    }
-    private def cssIE9(project: String): String = {
-      project match {
-        case "facia" => "stylesheets/ie9.head.facia.css"
-        case "identity" => "stylesheets/ie9.head.identity.css"
-        case "football" => "stylesheets/ie9.head.football.css"
-        case "index" => "stylesheets/ie9.head.index.css"
-        case _ => "stylesheets/ie9.head.content.css"
-      }
-    }
-  }
+}
 
-  object js {
-     private def inlineJs(path: String): String = IOUtils.toString(AssetFinder(path))
+object js {
+  val curl: String = Get(LoadFromClasspath("assets/curl-domReady.js").map(RelativePathEscaper.escapeLeadingDotPaths))
+  val omnitureJs: String = Get(LoadFromClasspath("assets/vendor/omniture.js"))
+  val analyticsJs: String = Get(LoadFromClasspath("assets/projects/common/modules/analytics/analytics.js"))
+}
 
-     val curl: String = RelativePathEscaper.escapeLeadingDotPaths(inlineJs("assets/curl-domReady.js"))
-     val omnitureJs: String = inlineJs("assets/vendor/omniture.js")
-     val analyticsJs: String =  inlineJs("assets/projects/common/modules/analytics/analytics.js")
+object Get {
+  def apply[T](`try`: Try[T]) = `try` match {
+    case Success(s) => s
+    case Failure(e) => throw e
   }
 }
 
-object AssetFinder {
-  def apply(assetPath: String): URL = {
-    Option(Play.classloader(Play.current).getResource(assetPath)).getOrElse {
-      throw AssetNotFoundException(assetPath)
+// gets the asset url from the classpath
+object LoadFromClasspath {
+  def apply(assetPath: String): Try[String] = {
+    (Option(Play.classloader(Play.current).getResource(assetPath)) match {
+      case Some(s) => Success(s)
+      case None => Failure(AssetNotFoundException(assetPath))
+    }).flatMap { url =>
+      Try(IOUtils.toString(url))
     }
   }
 }
 
-case class AssetNotFoundException(assetPath: String) extends Exception {
-  override val getMessage: String = if (Configuration.assets.useHashedBundles) {
-    s"Cannot find asset $assetPath. You should run `make compile`."
-  } else {
-    s"Cannot find asset $assetPath. You should run `make compile-dev`."
-  }
-}
+case class AssetNotFoundException(assetPath: String) extends Exception(s"Cannot find asset $assetPath. You should run `make compile`.")
```

Comment:
kind of, well the build should have done it on the real built artifacts, so it shouldn't really happen!  If it did happen we'd be in trouble.